File: leetcode/js/ant-on-the-boundary.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var returnToBoundaryCount = function (nums) {
  let result = 0;
  let steps = 0;

  for (const num of nums) {
    steps += num;

    if (steps === 0) {
      result++;
    }
  }

  return result;
};

File: leetcode/rust/ant-on-the-boundary.rs
impl Solution {
    pub fn return_to_boundary_count(nums: Vec<i32>) -> i32 {
        let mut result = 0;
        let mut steps = 0;

        for num in nums {
            steps += num;

            if steps == 0 {
                result += 1;
            }
        }

        result
    }
}
